src/common/transformations/src/transformations/op_conversions/eye_decomposition.cpp

```diff
@@ -0,0 +1,123 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/op_conversions/eye_decomposition.hpp"
+
+#include <memory>
+
+#include "itt.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/op/util/gather_nd_base.hpp"
+#include "openvino/op/util/op_types.hpp"
+#include "openvino/opsets/opset9.hpp"
+#include "openvino/pass/pattern/op/or.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+
+namespace ov {
+namespace pass {
+
+/** \brief Check if output is rank one and data type can be i32 or i64. */
+const auto is_rank_one_int_shape = [](const Output<Node>& output) -> bool {
+    return pattern::type_matches_any({element::i32, element::i64})(output) && pattern::rank_equals(1)(output);
+};
+
+/** \brief Predicate to check eye k node is valid. */
+const auto k_predicate = [](const Output<Node>& output) -> bool {
+    return is_rank_one_int_shape(output) && (output.get_partial_shape()[0].get_length() == 1);
```

Comment:
you need to check that this dimension is static, get_length() throws exception
![image](https://user-images.githubusercontent.com/21181928/185346869-1a5e3eab-a768-43a8-93e9-5c3a71438d79.png)